tests/unit/test_charm.py

```diff
@@ -176,6 +177,47 @@ def test_given_requirer_requested_certificate_when_certificate_available_then_ce
         self.assertEqual(requirers_certificates[0].csr, "test_csr")
         self.assertEqual(requirers_certificates[0].certificate, "test_cert")
 
+    def test_given_limit_to_one_request_set_when_second_certificate_requested_then_certificate_not_generated(  # noqa: E501
```

Comment:
I like that we test the whole mechanics here, but for future filters I think most of the testing should be done directly on the filter class, and only having one test that ensures the filter is properly loaded in the main charm.

Reply:
That would certainly make our job easier :D. I've added a example unit tests below also.